cpp/ql/src/semmle/code/cpp/commons/Printf.qll

```diff
@@ -846,15 +844,7 @@ class FormatLiteral extends Literal {
          len = 1
       or (
         this.getConversionChar(n).toLowerCase()="c" and
-        if (this.getEffectiveCharConversionChar(n)="C" and
-            not isMicrosoft() and
-            not isWideCharDefault()) then (
-          len = 6 // MB_LEN_MAX
-            // the wint_t (wide character) argument is converted
-            // to a multibyte sequence by a call to the wcrtomb(3) 
-        ) else (
-          len = 1 // e.g. 'a'
-        )
+        len = 1 // e.g. 'a'
```

Comment:
I tried to produce this case on several platforms and could never get more than one character to be output from a `%C` (or `%lc`).  Perhaps it's possible somehow, but it seems like a bad idea to use this logic to generate query results if it isn't accurate.